docs/types/reference-types.rst

```diff
@@ -504,13 +504,7 @@ Array slices are useful to ABI-decode secondary data passed in function paramete
         /// Forward call to "setOwner(address)" that is implemented by client
         /// after doing basic validation on the address argument.
         function forward(bytes calldata _payload) external {
-            // Since ABI decoding requires padded data, we cannot
-            // use abi.decode(_payload[:4], (bytes4)).
-            bytes4 sig =
-                _payload[0] |
-                (bytes4(_payload[1]) >> 8) |
-                (bytes4(_payload[2]) >> 16) |
-                (bytes4(_payload[3]) >> 24);
+            bytes4 sig = bytes4(_payload[:4]);
```

Comment:
Does the slice perform a length check? If yes, maybe just use `bytes4(_payload)` instead?

Reply:
Or maybe not. But in any case: Please write a comment about this here.

Reply:
It will not perform check, and there is a semantic test for that as well.
I would stick to more explicit version like `_payload[:4]`, but can add comment as well